rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/internal_interpreter.kt

```diff
@@ -375,13 +375,55 @@ class InternalInterpreter(val systemInterface: SystemInterface) {
                 is ZAddStmt -> {
                     assign(statement.target, eval(statement.expression))
                 }
+                is AddStmt -> {
+                    val addend1 = if (statement.left == null) {
```

Comment:
I would do this reasoning in the AddStmt class. You could do that while creating the object, changing the values it takes and how you calculate them, or using derived values. For example:

```
data class AddStmt(...) {
    @Derived
    val addend1 : Expression
          get() = if (statement.left...
```